contracts/savings/SavingsManager.sol

```diff
@@ -141,31 +181,95 @@ contract SavingsManager is ISavingsManager, PausableModule {
     function depositLiquidation(address _mAsset, uint256 _liquidated)
         external
         onlyLiquidator
+        whenStreamsNotFrozen
     {
         // transfer liquidated mUSD to here
         IERC20(_mAsset).safeTransferFrom(_liquidator(), address(this), _liquidated);
 
+        uint256 leftover = _allUnclaimedRewards(_mAsset);
+        _initialiseStream(_mAsset, _liquidated.add(leftover), DURATION);
+
+        emit LiquidatorDeposited(_mAsset, _liquidated);
+    }
+
+    /**
+     * @dev Collects the platform interest from a given mAsset and then adds capital to the
+     * stream. If there is > 24h left in current stream, just top it up, otherwise reset.
+     * @param _mAsset The mAsset to fetch interest
+     */
+    function collectAndStreamInterest(address _mAsset)
+        external
+        whenStreamsNotFrozen
+    {
+        ISavingsContract savingsContract = savingsContracts[_mAsset];
+        require(address(savingsContract) != address(0), "Must have a valid savings contract");
+
+        uint256 currentTime = now;
+        uint256 previousBatch = lastBatchCollected[_mAsset];
+        uint256 timeSincePreviousBatch = currentTime.sub(previousBatch);
+        require(timeSincePreviousBatch > 12 hours, "Cannot deposit twice in 12 hours");
+        lastBatchCollected[_mAsset] = currentTime;
+
+        // Batch collect
+        IMasset mAsset = IMasset(_mAsset);
+        (uint256 interestCollected, uint256 totalSupply) = mAsset.collectPlatformInterest();
+
+        if(interestCollected > 0){
+            // Validate APY
+            uint256 apy = _validateCollection(totalSupply, interestCollected, timeSincePreviousBatch);
+
+            // Get remaining rewards
+            uint256 leftover = _allUnclaimedRewards(_mAsset);
+
+            // Bundle this yield in with liquidator yield
+            // If (remainingTime < 24h)
+            //       take all remaining, add this, stream over next 24h
+            // else stream over remainingTime
+            uint256 end = streamEnd[_mAsset];
+            uint256 remaining = end > currentTime ? end.sub(currentTime) : 0;
+            uint256 newDuration = remaining < ONE_DAY ? ONE_DAY : remaining;
+            _initialiseStream(_mAsset, interestCollected.add(leftover), newDuration);
```

Comment:
In MIP-4 there are two different (contradictory?) durations: 1 week and 1 day. I assume 1 day is the correct one.

Reply:
1 day, yes. I will update the MIP